modules/zenith-public/lua/3-complete/hnm_respawn_windows.lua

```diff
@@ -0,0 +1,502 @@
+            -- PH logic: count kills, decide whether to pop HQ or re”pop NQ
+            local lastPop = GetServerVariable(config.popVar)
+            local kills   = GetServerVariable(config.phVar) + 1
+            local popNow = now > lastPop and
+                (math.random(1, 5) == 3 or kills > 6)
+
+            SetServerVariable(config.popVar, now)
+            SetServerVariable(config.phVar, kills)
+
+            local spawnId, blockId = popNow and hqId or nqId, popNow and nqId or hqId
+            DisallowRespawn(blockId, true)
+            DisallowRespawn(spawnId, false)
+            UpdateNMSpawnPoint(spawnId)
+            setAndPersistRespawnTime(GetMobByID(spawnId),
+                15)
+
+        elseif mobId == hqId then
+            -- HQ reset logic: push HQ out for 3 days, re”enable NQ immediately
+            SetServerVariable(config.popVar, now + 259200) -- 3 days
+            SetServerVariable(config.phVar, 0)
+
+            DisallowRespawn(hqId, true)
+            DisallowRespawn(nqId, false)
+            UpdateNMSpawnPoint(nqId)
+            setAndPersistRespawnTime(GetMobByID(nqId),
+                math.random(0, 6) * (5 * 60) + (21 * 3600))
+        end
+    end
+end
+
+-- Create beastmen NM overrides
+for _, config in pairs(beastmenNMs) do
+    m:addOverride(fmt('xi.zones.{}.Zone.onInitialize', config.zone),
+    function(zone)
+        local ID = zones[xi.zone[config.zoneId]]
+        super(zone)
+
+        local nqId = ID.mob[config.nqIdName]
+        local hqId = nqId + config.hqOffset
+        local currentTime = os.time()
+        -- first attempt NQ, then HQ; if neither has a saved respawn, roll a new NQ time
+        if
+            not processNqHqSpawn(nqId, hqId, config.nqName, currentTime) and
+            not processNqHqSpawn(hqId, nqId, config.hqName, currentTime)
+        then
+            restoreOrSetRespawnTime(nqId, math.random(0, 6) * (5 * 60) + (21 * 3600))
+        end
+    end)
+
+    -- NQ/HQ despawn override
+    for _, phase in ipairs{ config.nqName, config.hqName } do
+        m:addOverride(
+            fmt('xi.zones.{}.mobs.{}.onMobDespawn', config.zone, phase),
+            function(mob)
+                handleBeastmenDespawn(mob, config)
+            end
+        )
+    end
+end
+
+----------------------------------- SPECIAL CASE NMS -----------------------------------
+
+----------------------------------- Argus/Leech King Logic -----------------------------------
+local mazeOfShakhramiId = zones[xi.zone.MAZE_OF_SHAKHRAMI]
+
+-- pick one of the two NMs
+local function chooseArgusLK()
+    if math.random(1, 100) <= 50 then
+        return mazeOfShakhramiId.mob.ARGUS
+    else
+        return mazeOfShakhramiId.mob.LEECH_KING
+    end
+end
+
+-- common apply logic (either restoring or freshly setting)
+local function applyArgusLK(choice, restoreOnly)
+    local other = (choice == mazeOfShakhramiId.mob.ARGUS) and
+        mazeOfShakhramiId.mob.LEECH_KING or
+        mazeOfShakhramiId.mob.ARGUS
```

Comment:
```suggestion
    local other = mazeOfShakhramiId.mob.ARGUS
    if choice ~= mazeOfShakhramiId.mob.LEECH_KING then
        choice = mazeOfShakhramiId.mob.ARGUS
        other = mazeOfShakhramiId.mob.LEECH_KING
    end
```

I think this is more robust to handle the dichotomy (allows bunk to be sent to the function, even though that should be impossible) and be more readible

Reply:
changed to the suggested